pkg/http/http.go

```diff
@@ -62,25 +65,33 @@ func NewHTTPClient(config *HTTPClientConfig) (*HTTPClient, error) {
 		httpClient = http.DefaultClient
 	}
 
-	if config.Auth != nil && config.Auth.Type == "google_idtoken" {
+	if config.Auth != nil && (config.Auth.Type == "google_idtoken" || config.Auth.Type == "google_oauth2") {
 		var creds []byte
 		switch {
 		case config.Auth.CredentialsJSONBase64 != "":
-			v, err := base64.StdEncoding.DecodeString(config.Auth.CredentialsJSONBase64)
+			var err error
+			creds, err = decodeCredentials(config.Auth.CredentialsJSONBase64)
 			if err != nil {
-				return nil, fmt.Errorf("decoding credentials_json_base64: %w", err)
+				return nil, err
 			}
-			creds = v
 		default:
 			return nil, fmt.Errorf("missing credentials for google_idtoken auth")
 		}
 
 		ctx := context.Background()
-		ts, err := idtoken.NewTokenSource(ctx, config.Auth.Audience, idtoken.WithCredentialsJSON(creds))
-		if err != nil {
-			return nil, err
+		if config.Auth.Type == "google_idtoken" {
+			ts, err := idtoken.NewTokenSource(ctx, config.Auth.Audience, idtoken.WithCredentialsJSON(creds))
+			if err != nil {
+				return nil, err
+			}
+			httpClient = oauth2.NewClient(ctx, ts)
+		} else if config.Auth.Type == "google_oauth2" {
```

Comment:
Please add test case for covering google_oauth2 auth type

Reply:
Done